datastore/src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -1223,6 +1249,27 @@ private int blockOnDeletionJobGettingNumUpdates( UUID deletionJobId, boolean blo
         return 0;
     }
 
+    // Block until JobStatus is FINISHED or CANCELED
+    public void waitForDeleteJobToTerminate( UUID deletionJobId ) {
```

Comment:
This should be in job service. The most straight forward way is to introduce a job service topic and make the job service subscribe to it. Whenever a job is completed it fires a JobCompletedEvent that contains its job id and causes the service to release the lock held in a *local* Map<UUID, CountDownLatch>. Once the latch successfully counts down, remove the entry from the local map if it is still there.

Reply:
See ReliableTopic in Hazelcast.

Reply:
I added this primarily for testing. There is a rehearsal test that verifies that the data has actually been deleted, hence needs to wait until completion.